evil-owl.el

```diff
@@ -213,22 +232,65 @@ MARK points nowhere."
                            ((markerp pos) (marker-buffer pos)))))
     (with-current-buffer buffer
       (let ((line (line-number-at-pos pos))
-            (column (save-excursion
-                      (goto-char pos)
-                      (current-column))))
-        (list line column (current-buffer))))))
+            (display-string "")
+            column)
+        (save-excursion
+          (goto-char pos)
+          (setq column (current-column))
+          (when evil-owl-mark-context-lines
+            (let ((lines-traversed-upward 0)
+                  display-string-beg
+                  display-string-end)
+              (while (and (< lines-traversed-upward evil-owl-mark-context-lines)
+                          (not (bobp)))
+                (forward-line -1)
+                (setq lines-traversed-upward (+ 1 lines-traversed-upward)))
+              (setq display-string-beg (line-beginning-position))
+              (dotimes (_ (+ lines-traversed-upward
+                             evil-owl-mark-context-lines))
+                (forward-line 1))
+              (setq display-string-end (line-end-position))
+              (setq display-string
+                    (evil-owl--format-mark-display-string
+                     (buffer-substring
+                      display-string-beg
+                      display-string-end))))))
+        (list line column buffer display-string)))))
+
+(defun evil-owl--format-mark-display-string (display-string)
+  "Pad and truncate the DISPLAY-STRING shown in mark previews."
+  (let ((display-string-lines (split-string
+                               display-string "\n")))
+    (concat
+     "\n"
+     (mapconcat
+      (lambda (display-string-line)
+        (if (string-equal "" display-string-line)
```

Comment:
I prefer to special-case this since people might want to have `show-trailing-whitespace` on. Maybe there's a better way of doing this though.